osgp/protocol-adapter-iec60870/osgp-protocol-simulator-iec60870/src/main/java/org/opensmartgridplatform/simulator/protocol/iec60870/server/Iec60870ServerEventListener.java

```diff
@@ -0,0 +1,54 @@
+package org.opensmartgridplatform.simulator.protocol.iec60870.server;
+
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+import org.openmuc.j60870.Connection;
+import org.openmuc.j60870.ServerEventListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Iec60870ServerEventListener implements ServerEventListener {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Iec60870ServerEventListener.class);
+
+    private Iec60870ASduHandlerMap iec60870ASduHandlerMap;
+    private int connectionIdCounter = 1;
+    private int connectionTimeout;
+
+    public Iec60870ServerEventListener(final Iec60870ASduHandlerMap iec60870ASduHandlerMap,
+            final int connectionTimeout) {
+        this.iec60870ASduHandlerMap = iec60870ASduHandlerMap;
+        this.connectionTimeout = connectionTimeout;
+    }
+
+    @Override
+    public void connectionIndication(final Connection connection) {
+        final int connectionId = this.connectionIdCounter++;
+        LOGGER.info("Client connected on connection ({}).", connectionId);
+
+        try {
+            LOGGER.info("Waiting for StartDT on connection ({}) for {} ms.", connectionId, this.connectionTimeout);
+            connection.waitForStartDT(
+                    new Iec60870ConnectionEventListener(connection, connectionId, this.iec60870ASduHandlerMap),
+                    this.connectionTimeout);
+        } catch (final IOException | TimeoutException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1166)